includes/allDocsGenerator.js

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2023 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+const allDocsGenDebug = require('debug')('couchbackup:alldocsgenerator');
+const { BackupError } = require('./error.js');
+
+/**
+ * Async generator function for paginating _all_docs for shallow backups.
+ *
+ * @param {object} dbClient - object for connection to source database containing name, service and url
+ * @param {object} options - backup configuration
+ * @yields {object} a "done" type backup batch {command: d, batch: #, docs: [{_id: id, ...}, ...]}
+ */
+module.exports = async function * (dbClient, options = {}) {
+  let batch = 0;
+  let lastPage = false;
+  let startKey = null;
+  const opts = { db: dbClient.db, limit: options.bufferSize, includeDocs: true };
```

Comment:
This will need a tweak to work once it has the changes from #670
```suggestion
  const opts = { db: dbClient.dbName, limit: options.bufferSize, includeDocs: true };
```

Reply:
oh the test code might need tweaks for the request changes too